src/components/NewCommentForm.tsx

```diff
@@ -1,13 +1,12 @@
 import classNames from 'classnames';
 import React, { useState } from 'react';
-import { CommentData } from '../types/Comment';
+import { addNewComment } from '../features/Slices/comments';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
 
-type Props = {
-  onSubmit: (data: CommentData) => Promise<void>;
-};
-
-export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
+export const NewCommentForm: React.FC = () => {
   const [submitting, setSubmitting] = useState(false);
+  const selectedPost = useAppSelector(state => state.selectedPost.selectedPost);
```

Comment:
It seems like you are using a global state to hold the selected post. This might not be the best approach as it can lead to problems when the component is unmounted but the state is not cleaned up. You might want to consider using local state or passing the selected post as a prop instead.